### PR TITLE
[DB sources] : Exit when encountering a config error

### PR DESCRIPTION
**Pull request by akashkulk**
Currently, sources don't exit when encountering a config error, but instead default to the behavior of rethrowing. This causes `AirbyteExceptionHandler` to throw a system error thereby causing config errors to be reported to Sentry (potentially paging the OC rotation). 

Fix here is to treat config errors the same as transient errors and exit as soon as they are encountered

### CODE REVIEW

**Review by xiaohansong — APPROVED** [review]

**Review by evantahler — APPROVED** [review]

**Review by stephane-airbyte — APPROVED** [review]

File: airbyte-integrations/connectors/source-mysql/build.gradle [hunk 1]
```diff
@@ -8,7 +8,7 @@ plugins {
 airbyteJavaConnector {
     cdkVersionRequired = '0.35.11'
     features = ['db-sources']
-    useLocalCdk = false
+    useLocalCdk = true
```
Comment (theyueli) [inline airbyte-integrations/connectors/source-mysql/build.gradle:11]: not sure if we need to merge this change (and the one below) ?

Comment (akashkulk) [reply airbyte-integrations/connectors/source-mysql/build.gradle:11]: Will revert this when cdk is published and is toggled

File: airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunnerTest.kt [hunk 2]
```diff
@@ -221,32 +220,6 @@ internal class IntegrationRunnerTest {
         Mockito.verify(jsonSchemaValidator).validate(any(), any())
     }
 
-    @Test
-    @Throws(Exception::class)
-    fun testReadException() {
```
[the remark this answers is not in the document]
Comment (akashkulk) [reply airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunnerTest.kt:226]: This test fails now that we are properly exiting when catching the exception. We are changing the behavior of re-throwing the exception

**Review by theyueli — APPROVED** [review]

File: airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunnerTest.kt (hunk 2 above)
[the remark this answers is not in the document]
Comment (theyueli) [reply airbyte-cdk/java/airbyte-cdk/core/src/test/kotlin/io/airbyte/cdk/integrations/base/IntegrationRunnerTest.kt:226]: I see, in this case, we might be interested in using this JUnit rule to verify the exit status:

https://stefanbirkner.github.io/system-rules/#ExpectedSystemExit

**Pull request merged**